### PR TITLE
Add redis db backend

### PR DESCRIPTION
**Pull request by jiegec**

### CODE REVIEW

**Comment by bigeagle** [comment]
LGTM

**Comment by shankerwangmiao** [comment]
I wonder if we can adopt a database abstraction layer to reduce repeated code?

**Comment by jiegec** [comment]
> I wonder if we can adopt a database abstraction layer to reduce repeated code?

Yes, we can abstract one more level out here.

File: manager/server.go [hunk 1]
```diff
@@ -276,7 +276,7 @@ func (s *Manager) updateSchedulesOfWorker(c *gin.Context) {
 		curStatus, err := s.adapter.GetMirrorStatus(workerID, mirrorName)
 		s.rwmu.RUnlock()
 		if err != nil {
-			fmt.Errorf("failed to get job %s of worker %s: %s",
+			err = fmt.Errorf("failed to get job %s of worker %s: %s",
```
Comment (z4yx) [inline manager/server.go:279]: 这儿给 err 赋值后并没有用到吧。我想作者原意应该是打log，但是误用了函数。

**Review by z4yx — CHANGES_REQUESTED** [review]

File: manager/server.go (hunk 1 above)
Comment (jiegec) [reply manager/server.go]: 改成 logger.Errorf 了

**Review by z4yx — APPROVED** [review]

**Pull request merged**